### PR TITLE
Use `@typegoose/typegoose` instead of `@hasezoey/typegoose`

### PR DESCRIPTION
**Pull request by dirkbolte**
Addressing issue #33

### CODE REVIEW

**Comment by duro** [comment]
As mentioned in #33, I took this PR one step further and upgraded it to the new 6.0 version of typegoose. We should all discuss how best to release this.

Here is my PR into @dirkbolte's PR

https://github.com/dirkbolte/nestjs-typegoose/pull/1

**Comment by kpfromer** [comment]
@dirkbolte Once the pull request on your fork is merged with @duro 's fork could you pull on to master again. I think switching to `@typegoose/typegoose` with version `6.0.0` would be much more beneficial. Thanks for the request.

**Comment by dirkbolte** [comment]
@kpfromer Merged @duro 's PR.  This repo was forked from your latest commit - so this should be clean to merge now.

**Comment by olivmonnier** [comment]
Hi everyone, my little contribution:
https://github.com/kpfromer/nestjs-typegoose/blob/master/README.md#L79 to replace by `import { ModelType } from '@typegoose/typegoose/lib/types';`

File: src/typegoose.decorators.spec.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import {prop, Typegoose} from '@hasezoey/typegoose';
+import {prop, Typegoose} from '@typegoose/typegoose';
```
[the remark this answers is not in the document]
Comment (emjimadhu) [reply src/typegoose.decorators.spec.ts]: I can't suggest on line 9. It should be

```
class MockUser
```

File: src/typegoose.module.spec.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { TypegooseModule } from './typegoose.module';
 import { TypegooseCoreModule as CoreModule } from './typegoose-core.module';
-import { prop, Typegoose } from '@hasezoey/typegoose';
+import { prop, Typegoose } from '@typegoose/typegoose';
```
Comment (emjimadhu) [inline src/typegoose.module.spec.ts]:
```suggestion
import { prop } from '@typegoose/typegoose';
```

Comment (emjimadhu) [reply src/typegoose.module.spec.ts]: on line 7

```
class MockTask
```

And on line 11

```
class MockUser
```

File: test/app.e2e-spec.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import * as request from 'supertest';
 import { Test } from '@nestjs/testing';
 import { Body, Controller, Module, Post } from '@nestjs/common';
 import { InjectModel, TypegooseModule } from '../src';
-import { prop, Typegoose } from '@hasezoey/typegoose';
+import { prop, Typegoose } from '@typegoose/typegoose';
```
Comment (emjimadhu) [inline test/app.e2e-spec.ts]:
```suggestion
import { prop } from '@typegoose/typegoose';
```

Comment (emjimadhu) [reply test/app.e2e-spec.ts]: In Line 16

```
class MockTypegooseClass
```

File: README.md [hunk 4]
```diff
@@ -76,7 +76,7 @@ Get the cat model in a service
 import { Injectable } from "@nestjs/common";
 import { InjectModel } from "nestjs-typegoose";
 import { Cat } from "./cat.model";
-import { ModelType } from "@hasezoey/typegoose";
+import { ModelType } from "@typegoose/typegoose";
```
Comment (emjimadhu) [inline README.md]:
```suggestion

```

Comment (emjimadhu) [reply README.md]: And for line 83. I don't know how to change. According to doc we have to use `getModelForClass`.

Comment (kpfromer) [reply README.md]: I think that `getModelForClass` has been accounted for in `src/typegoose.providers.ts`.

Comment (dirkbolte) [reply README.md]: done

File: test/app.e2e-spec.ts (hunk 3 above)
Comment (dirkbolte) [reply test/app.e2e-spec.ts]: done

File: src/typegoose.module.spec.ts (hunk 2 above)
Comment (dirkbolte) [reply src/typegoose.module.spec.ts]: done

File: README.md [hunk 5]
```diff
@@ -39,7 +39,7 @@ Create class that extends [Typegoose](https://github.com/szokodiakos/typegoose#m
 **cat.model.ts**
 
 ```typescript
-import { prop, Typegoose } from "@hasezoey/typegoose";
+import { prop, Typegoose } from "@typegoose/typegoose";
```
[the remark this answers is not in the document]
Comment (dirkbolte) [reply README.md]: done

File: README.md [hunk 6]
```diff
@@ -303,7 +303,7 @@ The string given to `getModelToken` function should be the class name of the typ
 
 ## Requirements
 
-1.  @hasezoey/typegoose +5.9.2
+1.  @typegoose/typegoose +5.9.2
```
[the remark this answers is not in the document]
Comment (dirkbolte) [reply README.md]: done

File: src/typegoose.decorators.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dirkbolte) [reply src/typegoose.decorators.spec.ts]: done

File: README.md [hunk 7]
```diff
@@ -76,11 +76,11 @@ Get the cat model in a service
 import { Injectable } from "@nestjs/common";
 import { InjectModel } from "nestjs-typegoose";
 import { Cat } from "./cat.model";
-import { ModelType } from "@typegoose/typegoose";
+import { ReturnModelType } from "@typegoose/typegoose";
 
 @Injectable()
 export class CatsService {
-  constructor(@InjectModel(Cat) private readonly catModel: ModelType<Cat>) {}
+  constructor(@InjectModel(Cat) private readonly catModel: ReturnModelType<Cat>) {}
```
Comment (emjimadhu) [inline README.md]: Does `ReturnModelType` works properly? For me it was giving some error. May be i am using old `nestjs-typegoose` package? I am not sure! Please test this. Thanks.

**Comment by emjimadhu** [comment]
@kpfromer why it is not merged yet? It seems eveything looks great!

File: README.md (hunk 7 above)
Comment (dirkbolte) [reply README.md]: Right, there was one issue - has to be ```typeof Cat```

**Pull request merged**